matRad_calcParticleDose.m

```diff
@@ -160,22 +160,48 @@
 
             maxLateralCutoffDoseCalc = max(machine.data(energyIx).LatCutOff.CutOff);
 
-            % Ray tracing for beam i and ray j
-            [ix,radialDist_sq] = matRad_calcGeoDists(rot_coordsVdoseGrid, ...
+            if strcmp(anaMode, 'fineSampling')
+                % Ray tracing for beam i and ray j
+                [ix,~,~,~,latDistsX,latDistsZ] = matRad_calcGeoDists(rot_coordsVdoseGrid, ...
+                                                     stf(i).sourcePoint_bev, ...
+                                                     stf(i).ray(j).targetPoint_bev, ...
+                                                     machine.meta.SAD, ...
+                                                     find(~isnan(radDepthVdoseGrid{1})), ...
+                                                     maxLateralCutoffDoseCalc);
+                                                 
+                                                 
+                % evaluate initial sigma for every energy in ray
+                sigmaIni = matRad_calcSigmaIni(machine.data,stf(i).ray(j),stf(i).ray(j).SSD);
+
+                % Given the initial sigmas of the sampling ray, this
+                % function provides the weights for the sub-pencil beams,
+                % their positions and their sigma used for dose calculation
+                for k = 1:stf(i).numOfBixelsPerRay(j) % loop over all bixels per ray
+                    if (fineSamplingSigmaSub < sigmaIni(k)) && (fineSamplingSigmaSub > 0)
+                        [finalWeight(:,k), sigmaSub(:,k), posX(:,k), posZ(:,k), numOfSub(:,k)] = ...
+                                  matRad_calcWeights(sigmaIni(k), fineSamplingMethod, fineSamplingN, fineSamplingSigmaSub);
+                    else
+                        error('Problem with chosen sub beam sigma in fine sampling calculation!');
```

Comment:
Use matRad_cfg.dispError please if possible. Can you also specify the error better?